dex/elrond_dex_farm/src/farm_token_merge.rs

```diff
@@ -0,0 +1,180 @@
+            let entry = self.nft_deposit(caller).get(index);
+            require!(entry.token_id == farm_token_id, "Not a farm token");
+
+            tokens.push(FarmToken {
+                token_amount: entry.clone(),
+                attributes: self.get_farm_attributes(&entry.token_id, entry.token_nonce)?,
+            });
+
+            index += 1;
+        }
+
+        if replic.is_some() {
+            tokens.push(replic.unwrap());
+        }
+
+        if tokens.len() == 1 {
+            return Ok(tokens[0].clone().attributes);
+        }
+
+        let aggregated_attributes = FarmTokenAttributes {
+            reward_per_share: self.aggregated_reward_per_share(&tokens),
+            entering_epoch: self.aggregated_entering_epoch(&tokens),
+            apr_multiplier: self.aggregated_apr_multiplier(&tokens)?,
+            with_locked_rewards: self.aggregated_with_lock_rewards(&tokens)?,
+            initial_farming_amount: self.aggregated_initial_farming_amount(&tokens),
+            compounded_reward: self.aggregated_compounded_reward(&tokens),
+            current_farm_amount: self.aggregated_current_farm_amount(&tokens),
+        };
+
+        Ok(aggregated_attributes)
+    }
+
+    fn aggregated_reward_per_share(&self, tokens: &[FarmToken<Self::BigUint>]) -> Self::BigUint {
+        let mut dataset = Vec::new();
+        tokens.iter().for_each(|x| {
+            dataset.push(ValueWeight {
+                value: x.attributes.reward_per_share.clone(),
+                weight: x.token_amount.amount.clone(),
+            })
+        });
+        self.weighted_average_ceil(dataset)
+    }
+
+    fn aggregated_entering_epoch(&self, tokens: &[FarmToken<Self::BigUint>]) -> u64 {
+        let mut dataset = Vec::new();
+        tokens.iter().for_each(|x| {
+            dataset.push(ValueWeight {
+                value: Self::BigUint::from(x.attributes.entering_epoch),
+                weight: x.token_amount.amount.clone(),
+            })
+        });
+        let avg = self.weighted_average(dataset);
+        avg.to_u64().unwrap()
+    }
+
+    fn aggregated_apr_multiplier(&self, tokens: &[FarmToken<Self::BigUint>]) -> SCResult<u8> {
+        let first_elem_value = tokens.get(1).unwrap().attributes.apr_multiplier;
+        let mut same_value = true;
+        tokens
+            .iter()
+            .for_each(|x| same_value &= first_elem_value == x.attributes.apr_multiplier);
+        require!(same_value, "Cannot compute apr multiplier aggregate");
+        Ok(first_elem_value)
+    }
+
+    fn aggregated_with_lock_rewards(&self, tokens: &[FarmToken<Self::BigUint>]) -> SCResult<bool> {
+        let first_elem_value = tokens.get(1).unwrap().attributes.with_locked_rewards;
+        let mut same_value = true;
+        tokens
+            .iter()
+            .for_each(|x| same_value &= first_elem_value == x.attributes.with_locked_rewards);
+        require!(same_value, "Cannot compute with locked rewards aggregate");
```

Comment:
Why is it OK not to handle the case when there are tokens with different with_locked_rewards values?

Reply:
If you merge some tokens with different with_locked_rewards values and someone claims the rewards would mean you have to send some rewards and some locked_rewards. In order to keep things -not that complicated- I decided to avoid that case.

Reply:
Ok, thanks, that answers my question, but I would leave this as "unresolved" to remind me to check the implications of this require failing.

Reply:
Resolved